src/Common/QueryFuzzer.cpp

```diff
@@ -865,6 +937,35 @@ ASTs QueryFuzzer::getInsertQueriesForFuzzedTables(const String & full_query)
     return queries;
 }
 
+/// I am still not very expert in templates to reuse the code above
```

Comment:
Kind reminder that you don't need to solve everything with templates. The two functions are extremely similar. You could do it with templates, sure. But even without them, you could have an extra flag parameter that uses the optimize query or the insert one.

Reply:
I can try to clean this further